Switch to using openSUSE Leap 42.2

jayvdb opened the pull request:
Fixes https://github.com/coala/docker-coala-base/issues/144

jayvdb commented:
This is not ideal at the moment.  I dont mind if we loose GoTypeBear, but JuliaLintBear is a huge loss.  Hopefully the upstream problems are solved quickly.

Noting that we've lost Julia, this removes ~180 Mb from the image, down to 2.017 GB.
Re-adding Julia may mean there are much smaller size improvements.
However, I also havent optimised this at all -- no doubt there are additional packages which can be removed after switching to Leap. (e.g. the custom Node could be replaced with the system Node.)

jayvdb commented:
Note, this is close to finished, but there are a few things I would like to try.

sils commented:
@jayvdb sounds like you don't want a full review yet? Taking a quick look and marking WIP after. Mark it pending review when done trying a few things.

sils commented:
ack https://github.com/coala/docker-coala-base/pull/146/commits/699df2d3d0283813dc892b6c822f634a293c4137

sils commented on Dockerfile:
```diff
@@ -305,10 +305,11 @@ RUN mkdir -p ~/.RLibrary && \
 # Tailor (Swift) setup
+RUN curl -fsSL -o install.orig https://raw.githubusercontent.com/sleekbyte/tailor/master/script/install.sh
+
+RUN sed 's/read -r CONTINUE < \/dev\/tty/CONTINUE=y/' install.orig > install.sh
```
why move it up?

jayvdb commented:
For some reason, go-1.5 is being selected.  Probably need to pin something.

jayvdb commented:
Now busted by https://github.com/coala/docker-coala-base/issues/200

sils commented:
> Now busted by #200

What do you mean?

sils commented:
marking WIP because of conflicts and discussion at https://github.com/coala/docker-coala-base/issues/144#issuecomment-316653838

jayvdb committed:
Switch to using openSUSE Leap 42.2

Leap is more stable than Tumbleweed.
Downgrades lua from 5.3 to 5.2

Fixes https://github.com/coala/docker-coala-base/issues/144

jayvdb commented:
Current breakage is only https://github.com/coala/coala/issues/4519 , I believe.

jayvdb reviewed (commented):
I think it would be good to put the base repo url and string "openSUSE_Leap_42.2" into env variables, to make these lines a bit shorter and less repetitive.

jayvdb committed:
Dockerfile: Use python module ensurepip

Remove dependency on zypper package python3-pip, which
implicitly depends on python3-setuptools and other packages
which will be upgraded later in the build process.

Closes https://github.com/coala/docker-coala-base/issues/223

jayvdb committed:
Add libffi-devel

fneu commented:
I don't think openSUSE Leap 42.2 is up to date anymore
